### PR TITLE
feat(app): creates temp module slideout

### LINKED ISSUE #8790: Temperature Module Slide-out
# Overview
The temperature module slide-out can be accessed from the module card's action menu on both the device details page and the protocol run details page's run tab.

# Design
The design lives here: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A105540

# Implementation Details
- The accepted values for temp module temperature are 4ºC to 96ºC

# Acceptance Criteria

- [ ]  Setting an acceptable temp value and hitting save should update the saved value

### PR DESCRIPTION
**Pull request by jerader**
closes #8790 

# Overview

This create the temp module slideout and hooks it up

With button disabled:
Screen Shot 2022-02-09 at 2 38 18 PM

With button enabled:
Screen Shot 2022-02-09 at 2 38 42 PM

# Changelog

- creates `temperatureModuleSlideout`

# Review requests

- [ ] open in the app and have a temp module connected. In the module card, click on the overflow icon -> set temperature -> the slideout should open
- [ ] the slideout should display the title and the body with an input field in Celsius
- [ ] the slideout button should be disabled until a value is inputted in the field. 
- [ ] if you click on the button, the input field should be blank again with the button disabled

Note: the overflow menu stays opened even with the slideout rendered - this will be addressed in a follow up PR https://github.com/Opentrons/opentrons/issues/9413#issue-1128826297
# Risk assessment

low, behind ff

### CODE REVIEW

File: shared-data/js/constants.ts [hunk 1]
```diff
@@ -95,3 +95,5 @@ export const MAGNETIC_MODULE_V1_MAX_ENGAGE_HEIGHT = 40
 export const MAGNETIC_MODULE_V1_DISNEGAGED_HEIGHT = -5
 
 export const MAGNETIC_MODULE_TYPE_LABWARE_BOTTOM_HEIGHT = 0
+
+export const CELSIUS: '°C' = '°C'
```
Comment (sakibh) [inline shared-data/js/constants.ts:99]: Nice!

**Review by sakibh — APPROVED** [review]
Looks great! ✅

File: app/src/organisms/Devices/ModuleCard/TemperatureModuleSlideout.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          data-testid={`Temp_Slideout_input_field_${name}`}
+        >
+          <Text
+            fontWeight={FONT_WEIGHT_REGULAR}
+            fontSize={TYPOGRAPHY.fontSizeH6}
+            color={COLORS.darkGrey}
+          >
+            {t('temperature')}
+          </Text>
+          {/* TODO Immediately: make sure input field matches final designs */}
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/TemperatureModuleSlideout.tsx:77]: Looks like we're using the old input field that does not match up with the desiign reskin, which is fine, we should just figure out who's gonna make the new input field component and remember to come back and swap. or, if you feel like you have time now, you can make the generic input component from figma and plug it in here!

**Review by shlokamin — APPROVED** [review]
LGTM! As you already called out the input field does not match the new reskin so we just gotta figure out who's gonna do that and when. Also it looks like in figma the font color of the slideout description is black but ours is dark grey

**Comment by jerader** [comment]
> Also it looks like in figma the font color of the slideout description is black but ours is dark grey

@shlokamin oh weird, this much have changed since i made the magnetic module slideout. thanks for noticing - will fix this in the slideout component now

File: app/src/organisms/Devices/ModuleCard/TemperatureModuleSlideout.tsx (hunk 2 above)
Comment (jerader) [reply app/src/organisms/Devices/ModuleCard/TemperatureModuleSlideout.tsx:77]: was messing around a bit and i think this should be covered in a follow up ticket

**Pull request merged**